src/modules/UI/scenes/SendConfirmation/action.js

```diff
@@ -109,12 +114,24 @@ export const updateMaxSpend = () => (dispatch: Dispatch, getState: GetState) =>
 
 export const signBroadcastAndSave = () => async (dispatch: Dispatch, getState: GetState) => {
   const state = getState()
+  const account = getAccount(state)
   const selectedWalletId = getSelectedWalletId(state)
   const wallet = getWallet(state, selectedWalletId)
   const edgeUnsignedTransaction = getTransaction(state)
-  let edgeSignedTransaction = edgeUnsignedTransaction
+  const spendInfo = state.ui.scenes.sendConfirmation.spendInfo
+  if (!spendInfo) throw new Error('Invalid Spend Request')
```

Comment:
Should we be using an actual core error type here?

Reply:
Note that William said this does not have to come from the core since the core just spits out a boolean for whether the PIN is correct or not.

Reply:
there is no core error for this
im using it to make flow happy
perhaps we could start a single file in the gui that imports all the "core" errors, combines them with "gui" errors, and exports a single error object?

could be something like this
```
// errors.js
import { SomeCoreError } from 'edge-core-js'
const SpendInfoError = ...
const SomeOtherGuiError = ...

export  {
  SomeCoreError,
  SpendInfoError,
  SomeOtherGuiError
}

```
// some/other/file.js
import { SpendInfoError, SomeCoreError, SomeOtherGuiError } from '../../../errors.js

if (thingsGoSouth()) throw new SpendInfoError()
```